src/iterator.cr

```diff
@@ -1093,13 +1093,7 @@ module Iterator(T)
       if @tail
         tail_key, tail_val = @tail_key, @tail_val
         @tail = false
-        case tail_key
-        when nil, :_separator
-        when :_alone
-          return {tail_key.not_nil!, [tail_val]}
-        else
-          raise ArgumentError.new("symbols beginning with an underscore are reserved")
-        end
+        return {tail_key.not_nil!, [tail_val]} if tail_key == Enumerable::Chunk::Alone
```

Comment:
nil is hard here because, if i try to setup `previous = uninitialized U`, it gives error: 
` can't use Object as a Proc argument type yet, use a more specific type

    previous = uninitialized U
`